db/db_basic_test.cc

```diff
@@ -1287,6 +1287,45 @@ TEST_F(DBBasicTest, MultiGetBatchedMultiLevel) {
   }
 }
 
+TEST_F(DBBasicTest, MultiGetPrefixExtractor) {
+  Options options = CurrentOptions();
+  options.prefix_extractor.reset(NewFixedPrefixTransform(2));
+  BlockBasedTableOptions bbto;
+  bbto.filter_policy.reset(NewBloomFilterPolicy(10, false));
+  bbto.whole_key_filtering = false;
+  bbto.cache_index_and_filter_blocks = false;
+  options.table_factory.reset(NewBlockBasedTableFactory(bbto));
+  Reopen(options);
+
+  // First key is not in the prefix_extractor domain
+  ASSERT_OK(Put("k", "v0"));
+  ASSERT_OK(Put("kk1", "v1"));
+  ASSERT_OK(Put("kk2", "v2"));
+  ASSERT_OK(Put("kk3", "v3"));
+  ASSERT_OK(Put("kk4", "v4"));
+  ASSERT_OK(Flush());
+
+  std::vector<Slice> keys({"k", "kk1", "kk2", "kk3", "kk4"});
+  std::vector<PinnableSlice> values(keys.size());
+  std::vector<Status> s(keys.size());
+
+  db_->MultiGet(ReadOptions(), dbfull()->DefaultColumnFamily(), keys.size(),
+                keys.data(), values.data(), s.data(), true);
+
+  ASSERT_EQ(values.size(), keys.size());
+  ASSERT_EQ(std::string(values[0].data(), values[0].size()), "v0");
+  ASSERT_EQ(std::string(values[1].data(), values[1].size()), "v1");
+  ASSERT_EQ(std::string(values[2].data(), values[2].size()), "v2");
+  ASSERT_EQ(std::string(values[3].data(), values[3].size()), "v3");
+  ASSERT_EQ(std::string(values[4].data(), values[4].size()), "v4");
```

Comment:
would it be useful to add verification using stats or perf_context like bloom filter hits or misses?  so that we can know for sure that "It was incorrectly skipping keys not in the prefix_extractor domain, and not calling transform for keys in domain." is no longer the case?

Reply:
I structured the KVs such that if whole key filter lookup is used, the lookup will fail. But yes, it would be useful to add stats verification as it'll make it more robust to collisions etc.

Reply:
On second thoughts, it may not be that useful, since it won't tell us whether the bloom filter hit was due to prefix lookup or not.

Reply:
Turns out it was useful after all, as I wasn't updating the perf_context stats !